File: rb/features.rb
# ColoradoInformationMarketplace SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module ColoradoInformationMarketplaceFeatures
  def self.make_feature(name)
    case name
    when "base"
      ColoradoInformationMarketplaceBaseFeature.new
    when "test"
      ColoradoInformationMarketplaceTestFeature.new
    else
      ColoradoInformationMarketplaceBaseFeature.new
    end
  end
end

File: py/coloradoinformationmarketplace_sdk/features.py
# ColoradoInformationMarketplace SDK feature factory

from coloradoinformationmarketplace_sdk.feature.base_feature import ColoradoInformationMarketplaceBaseFeature
from coloradoinformationmarketplace_sdk.feature.test_feature import ColoradoInformationMarketplaceTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ColoradoInformationMarketplaceBaseFeature(),
        "test": lambda: ColoradoInformationMarketplaceTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
